README.md

```diff
@@ -1,2 +1,180 @@
+* **cacheId** - (String) An optional ID to be prepended to caches used by workbox-sw. This is primarily useful for local development where multiple sites may be served from the same http://localhost origin. Defaults to your app name (config.APP.name).
+* **runtimeCaching** - (Array of Object) Passing in an array of objects containing urlPatterns, handlers, and potentially options that will add the appropriate code to the generated service worker to handle runtime caching. The handler values correspond the names of the [strategies](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-sw.Strategies) supported by workbox-sw (cacheFirst, cacheOnly, networkFirst, networkOnly, staleWhileRevalidate)
+```JavaScript
+runtimeCaching: [{
+  // You can use a RegExp as the pattern:
+  urlPattern: /https://api.example.com/,
+  handler: 'cacheFirst',
+  // Any options provided will be used when
+  // creating the caching strategy.
+  options: {
+    cacheName: 'my-api-cache',
+    cacheExpiration: {
+      maxEntries: 10,
+    },
+  },
+},
+ ...
+]
+```
+* **maximumFileSizeToCacheInBytes** - (number) This value can be used to determine the maximum size of files that will be precached
+* **more** - For more details on Workbox configuration take a look at: [Workbox Google Developers](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-build)
+
+**Subscribe to events:**
+
+ServiceWorkerService triggers the following events:
+
+  - `registrationComplete`: sw successfully registered
+  - `registrationError`: sw not registered
+  - `newSWActive`: new sw controlling page
+  - `newSWWaiting`: new sw waiting for controlling page
+  - `unregistrationComplete`: all sw are unregistered
+
+**Why and how to use this events?**
+
+By default, users have to close all tabs to a site in order to update a Service Worker. The Refresh button is not enough.
+If you make a mistake here, users will see an outdated version of your site even after refreshing
+Service Workers break the Refresh button because they behave like “apps,” refusing to update while the app is still running, in order to maintain code consistency and client-side data consistency. We can write code to notify users when a new version is available. 
+
+This addon make it easy for you and implements [google recommendation](https://developers.google.com/web/tools/workbox/guides/advanced-recipes#offer_a_page_reload_for_users)
+Basically, what you have to do is subscribing to the event "newSWWaiting". When event is triggered, send a message to sw in order to launch skipWaiting + clients.claim on it to turn it active (you can do this just calling forceActivate method on serviceWorkerService). When service worker became active it will send a message "reload-window" and "newSWActive" will be triggered.
```

Comment:
What events names and code in its own tags:

- "newSWWaiting" => `newSWWaiting`
- skipWaiting + clients.claim => `skipWaiting + clients.claim`